block-storage/src/main/scala/coop/rchain/blockstorage/util/io/RandomAccessIO.scala

```diff
@@ -0,0 +1,71 @@
+package coop.rchain.blockstorage.util.io
+
+import java.io.{FileNotFoundException, IOException, RandomAccessFile}
+import java.nio.file.Path
+
+import cats.data.EitherT
+import cats.effect.Sync
+import cats.implicits._
+import IOError.IOErrT
+
+import scala.language.higherKinds
+
+final case class RandomAccessIO[F[_]: Sync](file: RandomAccessFile) {
+  def close(): IOErrT[F, Unit] =
+    EitherT(Sync[F].delay { file.close() }.attempt).leftMap[IOError] {
```

Comment:
`EitherT(Sync[F].delay { action }.attempt).leftMap[IOError]` is repeated like a lot. maybe it would make sense to figure out either what @goral09 suggested or at least extract it to a helper?

Reply:
I would prefer a different approach without `Either` and `EitherT`. The main reason for this mapping is because `IOException` is not specific enough. We are mapping to a more specific error. `Sync` extends `ApplicativeError` for `Throwable`. Why not use `Sync[F].raiseError` here? However, `IOError` would have to extend `Throwable` to satisfy the error type. But the return type could then be `F[Unit]`. The layer above this functions could then take a happy path approach, call `attempt` over the whole program and then map the left side to `IOError` (with a partial function).

Reply:
What I don't like with the 
```scala
def openFile[F[_]: MonadError[?[_], Error]](file: Path): F[Unit] = ???
```
the approach is that type system doesn't enforce (or at least points at) how the error should be handled because it [error] is now hidden inside `MonadError` and caller of this method has to extract it explicitly. With opposed to:
```scala
def openFile[F[_]](file: Path): F[Either[Error, Unit]] = ???
```
where an error is explicit and the caller has to map over the returned values explicitly. He [caller] can then make decision to ignore the error (by simple mapping `F[Either[E, Unit]]` to `F[Unit]` or handle it and then map. 

I am curious to hear your opinion on that.

Reply:
@sebtomba I undestand your point that there are still 2 error channels but @rabbitonweb mentioned in our [previous conversation](https://github.com/rchain/rchain/pull/2076#discussion_r247084165) that `Sync[F].raiseError` should only be used for fatal errors and agree with him on that. Since I believe this kind of IO errors could be handled by, for example, block storage in some meaningful way I decided to return explicit errors here instead of throwing them with `Sync[F]`.

Your and @goral09's point on returning `F[Either[IOErr, Unit]]` is completely reasonable though.